Final Model.ipynb

```diff
@@ -74,134 +29,1297 @@
     }
    ],
    "source": [
-    "features = []\n",
-    "for index, row in dataset.iterrows():\n",
-    "    print(index)\n",
-    "    track = row[\"track\"]\n",
-    "    track = track.replace(\" \", \"\")\n",
-    "    \n",
-    "    artist = row[\"artist\"]\n",
-    "    artist = artist.replace(\" \", \"\")\n",
-    "    \n",
-    "    title = track + \"-\" + artist\n",
-    "    \n",
-    "    audio_path = os.path.join(\"audio\", title + \".wav\")\n",
-    "    \n",
-    "    y, sr = librosa.load(audio_path)\n",
-    "\n",
-    "    # Mel-frequency Cepstral coefficients\n",
-    "    mfccs = np.mean(librosa.feature.mfcc(y=y, sr=sr, n_mfcc=13), axis=1)\n",
-    "    # Chroma features\n",
-    "    chroma = np.mean(librosa.feature.chroma_stft(y=y, sr=sr), axis=1)\n",
-    "    # Spectral centroid\n",
-    "    spectral_centroid = np.mean(librosa.feature.spectral_centroid(y=y, sr=sr))\n",
-    "    # Spectral bandwith\n",
-    "    spectral_bandwidth = np.mean(librosa.feature.spectral_bandwidth(y=y, sr=sr))\n",
-    "    # Spectral contrast\n",
-    "    spectral_contrast = np.mean(librosa.feature.spectral_contrast(y=y, sr=sr))\n",
-    "    # Spectral rolloff\n",
-    "    spectral_rolloff = np.mean(librosa.feature.spectral_rolloff(y=y, sr=sr))\n",
-    "    # Tempo\n",
-    "    tempo, _ = librosa.beat.beat_track(y=y, sr=sr)\n",
-    "    # Zero crossing rate\n",
-    "    zero_crossing_rate = np.mean(librosa.feature.zero_crossing_rate(y))\n",
-    "\n",
-    "    features.append(np.hstack([mfccs, chroma, spectral_centroid, spectral_bandwidth, \n",
-    "                          spectral_contrast, spectral_rolloff, tempo, zero_crossing_rate]))\n",
-    "    \n",
-    "    \n",
-    "mfcc_headers = [f'mfcc_{i}' for i in range(13)]\n",
-    "chroma_headers = [f'chroma_{i}' for i in range(12)]\n",
-    "\n",
-    "columns = []\n",
-    "columns.extend(mfcc_headers)\n",
-    "columns.extend(chroma_headers)\n",
-    "columns.extend(['spectral_centroid', 'spectral_bandwidth', 'spectral_contrast', 'spectral_rolloff',\n",
-    "              'tempo', 'zero_crossing_rate'])\n",
-    "\n",
-    "feature_df = pd.DataFrame(features, columns=columns)\n",
-    "\n",
-    "feature_df.to_csv('sound_features.csv', index=False)\n"
+    "complete_dataset.head()"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 31,
+   "execution_count": 9,
+   "id": "34766a65-e849-469a-bd23-91b75eab5c35",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "complete_dataset['track'] = complete_dataset['track'].str.replace(r'\\s+', '', regex=True)\n",
+    "complete_dataset['artist'] = complete_dataset['artist'].str.replace(r'\\s+', '', regex=True)\n",
+    "complete_dataset['file'] = complete_dataset['track'] + '-' + complete_dataset['artist']\n",
+    "complete_dataset.drop(['track', 'artist'], axis=1, inplace=True)\n",
+    "complete_dataset = complete_dataset[['file'] + [col for col in complete_dataset.columns if col != 'file']]\n"
```

Comment:
How is this changing it?

Reply:
We stored the unique key of each track differently across the sound features, spectrograms, etc. This code just makes it consistent so that we can use it as a key.